### PR TITLE
 [fix #937] Hide country input if SMS not selected. 

### LINKED ISSUE #937: Country required for Non-SMS S2D Snippets
Hi @glogiotatidis. 

I am updating Send to Device snippets for the F100 Snippet experience. I notice the new Snippet Admin tool requires a country code for all Send to Device snippets, even when SMS is not checked off.

Right now, when left blank, the country code defaults to `US`. And when I remove it, I get an error notice. This is even when SMS is unchecked. Perhaps this requirement needs to be conditional when SMS capabilites are checked. 
 

We have two types of send to device snippets.
- SMS and Email
-Email Only

Since we don't support SMS in all locales, we allow users to email themselves a download link. The requirement of a country code, prevents us from reaching all users and would require us to create a snippet for each country

Please see the example here: 
https://snippets-admin.mozilla.org/admin/base/asrsnippet/11428/change/?_changelist_filters=q%3D_week12_%2B_de_

SnippetID: 11428
Locale: DE

See screenshots.

Screen Shot 2019-03-28 at 11 30 23 AM

Screen Shot 2019-03-28 at 11 21 23 AM

Thank you!

### PR DESCRIPTION
**Pull request by jpetto**
UI only approach to the problem. this might be good enough?

@glogiotatidis thoughts?

### CODE REVIEW

**Comment by glogiotatidis** [comment]
It's OK for now. Make sure to list Country after the include SMS checkbox for better UX. Thanks!

File: snippets/base/static/js/admin/sms-country-toggle.js [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+            const country_default = country_input.value;
+            
+            function displayCountryField() {
+                country_row.classList.add('visible');
+            }
+
+            function hideCountryField() {
+                country_row.classList.remove('visible');
+                
+                // set the value back to default if user left it empty
```
Comment (glogiotatidis) [inline snippets/base/static/js/admin/sms-country-toggle.js]: This doesn't seem to work, field remains empty and raises an error when saving

Comment (jpetto) [reply snippets/base/static/js/admin/sms-country-toggle.js]: hm, can you paste the error you're getting? it's working as expected on my end.

Comment (glogiotatidis) [reply snippets/base/static/js/admin/sms-country-toggle.js]: I get a generic "Please correct the error bellow" with no errors displayed, b/c the field is hidden. If I check the SMS option, I see the field and it's empty and the error is "This field is required".

STR:
 1. Create a new S2D snippet
 2. Fill in all required fields
 3. Make sure to check include sms
 4. Empty country field
 5. Save
 6. Snippet does not save, correctly, due to empty country field
 7. Uncheck `include sms`
 8. Save
 9. Snippet still doesn't save, incorrectly this time b/c country field is not visible and should default to `us`.

**Comment by glogiotatidis** [comment]
Remember to update the commit messages!

File: snippets/base/static/js/admin/sms-country-toggle.js (hunk 1 above)
Comment (jpetto) [reply snippets/base/static/js/admin/sms-country-toggle.js]: hm, i believe i have this fixed. however, i cannot create a snippet, as the first `Locale` drop down has no options (and is required).

how can i populate that select box? i've run the commands in the readme (`create_locales()`) and see locale data in the db (`TargetedLocale.objects.all()`).

Comment (jpetto) [reply snippets/base/static/js/admin/sms-country-toggle.js]: i figured it out while working on #996. 😄 followed the STR above and was able to successfully save the snippet.

**Comment by glogiotatidis** [comment]
lgtm, thanks! :cake:

**Pull request merged**